src/core/managers/toolManager.js

```diff
@@ -319,112 +312,100 @@ class ToolManager {
+                    continue;
                 }
+            }
 
-                // Add the modified files to Git staging area
-                const addResult = await gitUtils.addFiles(modifiedFiles);
-
-                if (!addResult.success) {
-                    this.logger.warn(`Git add failed: ${addResult.error}`, 'Git auto-commit');
-                    return;
-                }
+            // Only create snapshot if we have files to backup
+            if (existingFiles.size > 0) {
+                await snapshotManager.createSnapshot(instruction, existingFiles, {
+                    triggerType: 'tool_execution',
+                    toolName,
+                    filePaths,
+                });
+            }
+        } catch (error) {
+            // Log error but don't fail tool execution
+            console.warn(`Failed to create pre-execution snapshot: ${error.message}`);
+        }
+    }
 
-                // Check status again after adding files
-                const statusAfterAdd = await gitUtils.getStatus();
-                if (statusAfterAdd.success) {
-                    this.logger.debug(
-                        `Git status after add: ${statusAfterAdd.status}`,
-                        'Git auto-commit'
-                    );
-                }
+    /**
+     * Check if a tool modifies files
+     * @private
+     * @param {string} toolName - Name of the tool
+     * @returns {boolean} True if tool modifies files
+     */
+    _isFileModifyingTool(toolName) {
+        const fileModifyingTools = [
+            'write_file',
+            'edit_file',
+            // Add more file-modifying tools as needed
+        ];
+
+        return fileModifyingTools.includes(toolName);
+    }
 
-                // Then commit the changes
-                const commitResult = await snapshotManager.commitChangesToGit(modifiedFiles);
-                if (!commitResult.success) {
-                    // Show full error message with proper formatting
-                    this.logger.warn('Git commit failed:', 'Git auto-commit');
-                    const errorLines = commitResult.error.split('\n');
-                    errorLines.forEach(line => {
-                        if (line.trim()) {
-                            this.logger.warn(`  ${line}`, '');
-                        }
-                    });
-                }
-            } catch (error) {
-                this.logger.warn(`Git commit error: ${error.message}`, 'Git auto-commit');
-                if (error.stack) {
-                    this.logger.debug(`Stack trace: ${error.stack}`);
+    /**
+     * Extract file paths from tool arguments
+     * @private
+     * @param {string} toolName - Name of the tool
+     * @param {Object} toolArgs - Tool arguments
+     * @returns {string[]} Array of file paths that will be modified
+     */
+    _extractFilePathsFromArgs(toolName, toolArgs) {
+        const filePaths = [];
```

Comment:
it should be a tool that knows if it is modyfying somethign and what, there should be common interface